src/lib/operators/table_scan/abstract_dereferenced_column_table_scan_impl.cpp

```diff
@@ -71,6 +76,20 @@ void AbstractDereferencedColumnTableScanImpl::_scan_reference_segment(const Refe
       matches[match_idx].chunk_offset = sub_pos_list.original_positions[matches[match_idx].chunk_offset];
     }
   }
+
+  if (predicate_condition != PredicateCondition::IsNull) {
+    return;
+  }
+
+  const auto& sub_pos_list = chunk_offsets_by_chunk_id[referenced_chunk_count];
+  const auto& position_filter = sub_pos_list.row_ids;
+
+  if (!position_filter || position_filter->empty()) {
+    return;
+  }
+  for (auto& pos : *position_filter) {
```

Comment:
Pos can be const. You could deref `position_filter` already in line 85 and use `.empty()` and no * in the loop (shouldn't matter, but maybe easier to read).

You could add comments on why you exit early in line 79 and what happens if you don't exit early. Plus explain that an access to `chunk_offsets_by_chunk_id[referenced_chunk_count]` (and not `referenced_chunk_count-1`) is done on purpose (looks wrong if you are not aware how splitting works in this case).

Reply:
As we know the size of the position filter (maybe also rename it to something like `remaining_null_row_id_positions` or the `sub_pos_list` to something like that), we should reserve.

Reply:
How do you like it now?